be/src/storage/lake/lake_persistent_index.cpp

```diff
@@ -65,10 +144,36 @@ Status LakePersistentIndex::try_replace(size_t n, const Slice* keys, const Index
         }
     }
     RETURN_IF_ERROR(_memtable->replace(keys, values, replace_idxes, _version.major_number()));
+    if (is_memtable_full()) {
+        RETURN_IF_ERROR(minor_compact());
+        flush_to_immutable_memtable();
+    }
     return Status::OK();
 }
 
 Status LakePersistentIndex::minor_compact() {
+    if (_immutable_memtable == nullptr) {
```

Comment:
is this a real compaction op? the code dump the immutable memtable to file and add it into _sstables, looks nothing to do with compaction?

Reply:
The definition of minor compaction is flush memtable to sstable.

Reply:
then why directly call it persistmemtable or dumptosstable?

Reply:
minor_compact is referred to https://leveldb-handbook.readthedocs.io/zh/latest/compaction.html

Reply:
ok